src/components/structure/visualizing/artefact-card/artefact-card.tsx

```diff
@@ -14,16 +15,34 @@ type Props = {
   imgAlt?: string,
   classification?: string,
 }
+
+
 const ArtefactCard: FC<Props> = ({
-  id='',
+  id = '',
   title = '',
   subtitle = '',
   date = '',
   to = '',
   imgSrc = '',
   imgAlt = '',
   classification = '',
-}) => (
+}) => {
+
+  const { collection } = useContext(StoreContext);
+
+  let isStoredFavorite = collection?.artefacts.includes(id);
+  const [isFavorite, setFavorite] = useState(!!isStoredFavorite);
+
+  const toggleFav = () => {
+    if (isFavorite) {
```

Comment:
Ok, ich glaube ich verstehe deinen Gedankengang nun besser.

Hatte vergessen zu erwähnen, dass eine Komponente, in der man den StoreContext nutzt und auf Store-Daten zugreift mit einem Aufruf des MobX-Observer-Hooks "umhüllt" werde muss.

Das `export default observer(...);` am Ende eine Komponente.

Das DIFF dazu:

```diff
diff --git a/src/components/structure/visualizing/artefact-card/artefact-card.tsx b/src/components/structure/visualizing/artefact-card/artefact-card.tsx
index fbf3b1b..cd0bc50 100644
--- a/src/components/structure/visualizing/artefact-card/artefact-card.tsx
+++ b/src/components/structure/visualizing/artefact-card/artefact-card.tsx
@@ -1,5 +1,7 @@
 import React, { FC, useContext, useState } from 'react';
 
+import { observer } from 'mobx-react-lite';
+
 import Image from '../../../base/visualizing/image';
 import StoreContext from '../../../../store/StoreContext';
 
@@ -27,19 +29,16 @@ const ArtefactCard: FC<Props> = ({
   imgAlt = '',
   classification = '',
 }) => {
-
   const { collection } = useContext(StoreContext);
 
-  let isStoredFavorite = collection?.artefacts.includes(id);
-  const [isFavorite, setFavorite] = useState(!!isStoredFavorite);
+  let isStoredFavorite = !!(collection?.artefacts.includes(id));
 
   const toggleFav = () => {
-    if (isFavorite) {
+    if (isStoredFavorite) {
       collection?.removeArtefactFromCollection(id);
     } else {
       collection?.addArtefactToCollection(id);
     }
-    setFavorite(!isFavorite);
   };
 
   return(
@@ -66,7 +65,7 @@ const ArtefactCard: FC<Props> = ({
               <p className="artefact-card__smalltext">{id}</p>
             </a>
             <a
-              className={`artefact-card__favorite ${isFavorite ? 'artefact-card__favorite--is-active' : ''}` }
+              className={`artefact-card__favorite ${isStoredFavorite ? 'artefact-card__favorite--is-active' : ''}` }
               onClick={toggleFav}
             ></a>
           </div>
@@ -76,4 +75,4 @@ const ArtefactCard: FC<Props> = ({
   );
 };
 
-export default ArtefactCard;
+export default observer(ArtefactCard);
```

Deswegen hat das Toggling wohl anfänglich nicht geklappt.
Hatte übersehen, dass am Ende der Komponente kein `export default observer(ArtefactCard)` stand.

Reply:
Alles klar :)